### PR TITLE
Add `.gitkeep` as an ignored filename

### LINKED ISSUE #643: Add `.gitkeep` as one of default valid filenames
With the change from https://github.com/twisted/towncrier/pull/622, a [very common filename `.gitkeep`](https://github.com/search?q=gitkeep&type=code) was omitted from the list of valid filenames.

Would you please consider adding the filename `.gitkeep` to the list at https://github.com/MetRonnie/towncrier/blob/9b685033b22bd3ee1bd81955a333e0e20bef02e6/src/towncrier/_builder.py#L117?

I would be happy to submit a PR to save myself several hours of work to modify our repositories in the Plone organization alone.

Thank you for your consideration.

Ref https://github.com/plone/volto/issues/6225

### LINKED ISSUE #659: Release 24.8.0
Just to track the new release

### PR DESCRIPTION
**Pull request by stevepiercy**
# Description

Fixes #643.

References #622.

# Checklist
* [x] Make sure changes are covered by existing or new tests.
* [x] For at least one Python version, make sure test pass on your local environment.
* [x] Create a file in `src/towncrier/newsfragments/`. Briefly describe your
  changes, with information useful to end users. Your change will be included in the public release notes.
* [ ] Make sure all GitHub Actions checks are green (they are automatically checking all of the above).
* [x] Ensure `docs/tutorial.rst` is still up-to-date.
* [x] If you add new **CLI arguments** (or change the meaning of existing ones), make sure `docs/cli.rst` reflects those changes.
* [x] If you add new **configuration options** (or change the meaning of existing ones), make sure `docs/configuration.rst` reflects those changes.

### CODE REVIEW

**Review by adiroiban — COMMENTED** [review]
Thanks for the update. It looks good.
I don't think that we need to update the test but maybe it would help to update the documentation to mention the new ignored file name.

**Comment by stevepiercy** [comment]
Docs updated in [d47c62e](https://github.com/twisted/towncrier/pull/647/commits/d47c62eb42fba6871c9c6acab7a31486054edce6).

File: docs/configuration.rst [hunk 1]
```diff
@@ -133,11 +133,20 @@ Top level keys
 ``ignore``
     A case-insensitive list of filenames in the news fragments directory to ignore.
 
+    ``None`` by default.
+
     ``towncrier check`` will fail if there are any news fragment files that have invalid filenames, except for those in the list. ``towncrier build`` will likewise fail, but only if this list has been configured (set to an empty list if there are no files to ignore).
 
-    Some filenames such as .gitignore, README.rst. README.md, and the template file, are automatically ignored.
+    The following filenames are automatically ignored.
```
Comment (adiroiban) [inline docs/configuration.rst]:
```suggestion
    The following filenames are automatically ignored, case insensitive.
```

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**